va/caa_test.go

```diff
@@ -589,27 +589,57 @@ func TestCAAFailure(t *testing.T) {
 }
 
 func TestParseResults(t *testing.T) {
+	// An empty slice of caaResults should return nil, "", nil
 	r := []caaResult{}
-	s, records, err := parseResults(r)
+	s, response, err := parseResults(r)
 	test.Assert(t, s == nil, "set is not nil")
 	test.Assert(t, err == nil, "error is not nil")
-	test.Assert(t, records == nil, "records is not nil")
-	test.AssertNotError(t, err, "no error should be returned")
-	r = []caaResult{{nil, errors.New("")}, {[]*dns.CAA{{Value: "test"}}, nil}}
-	s, records, err = parseResults(r)
+	test.Assert(t, response == "", "records is not nil")
+	// A slice of empty caaResults should return nil, "", nil
+	r = []caaResult{
+		{[]*dns.CAA{}, "", nil},
+		{[]*dns.CAA{}, "", nil},
+		{[]*dns.CAA{}, "", nil},
+	}
+	s, response, err = parseResults(r)
+	test.Assert(t, s == nil, "set is not nil")
+	test.Assert(t, err == nil, "error is not nil")
+	test.AssertEquals(t, response, "")
+	// A slice of caaResults containing an error followed by a CAA
+	// record should return the error
+	r = []caaResult{
+		{nil, "", errors.New("")},
+		{[]*dns.CAA{{Value: "test"}}, "", nil},
+	}
+	s, response, err = parseResults(r)
 	test.Assert(t, s == nil, "set is not nil")
 	test.AssertEquals(t, err.Error(), "")
-	expected := dns.CAA{Value: "other-test"}
-	test.AssertEquals(t, len(records), 0)
-	r = []caaResult{{[]*dns.CAA{&expected}, nil}, {[]*dns.CAA{{Value: "test"}}, nil}}
-	s, records, err = parseResults(r)
+	test.AssertEquals(t, response, "")
+	//  A slice of caaResults containing an error sandwiched between CAA
+	//  records should return the first CAA record
+	expected := dns.CAA{Value: "foo"}
+	r = []caaResult{
+		{[]*dns.CAA{&expected}, "foo", nil},
+		{nil, "", errors.New("")},
+		{[]*dns.CAA{{Value: "test"}}, "", nil},
```

Comment:
Technically this row isn't necessary; this case is less about "sandwiched" and more about "good record precedes error".